gomock/stack.go

```diff
@@ -0,0 +1,43 @@
+package gomock
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+const skipFrames = 2
+
+type stackTracer interface {
+	StackTrace() errors.StackTrace
+}
+
+func stackTraceStringFromError(err error, skipFrames int) string {
+	if err, ok := err.(stackTracer); ok {
+		return stackTraceString(err.StackTrace(), skipFrames)
+	}
+	return ""
+}
+
+func currentStackTrace(skipFrames int) string {
+	err := errors.New("fake error just to get stack")
+	if err, ok := err.(stackTracer); ok {
+		return stackTraceString(err.StackTrace(), skipFrames)
+	}
+	return ""
+}
+
+func stackTraceString(stackTrace errors.StackTrace, skipFrames int) string {
+	buffer := bytes.NewBufferString("")
```

Comment:
nit: It doesn't look like we need to use the function `bytes.NewBufferString("")` as we're taking in an empty string. Maybe just `bytes.Buffer{}`?